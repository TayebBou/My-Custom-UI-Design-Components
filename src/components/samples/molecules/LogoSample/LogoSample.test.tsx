import { render, screen } from '@testing-library/react'
import LogoSample from './LogoSample'
import { BrowserRouter as Router } from 'react-router-dom'

describe('LogoSample component', () => {
  beforeEach(() => {
    // Arrange
    render(
      <Router>
        <LogoSample />
      </Router>,
    )
  })
  test('A logo icon is rendered in the dom', () => {
    const logoIcon = screen.getByAltText('logo')
    // Assert
    expect(logoIcon).toBeInTheDocument()
  })
  test('A logo text is rendered in the dom', () => {
    const logoText = screen.getByText('UI Design Components')
    // Assert
    expect(logoText).toBeInTheDocument()
  })
})

export {}
