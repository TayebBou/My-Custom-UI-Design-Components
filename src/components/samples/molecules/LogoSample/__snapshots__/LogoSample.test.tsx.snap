// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoSample component should render correctly 1`] = `
<DocumentFragment>
  <h2>
    Logo Component
  </h2>
  <a
    class="logo logo"
    href="/"
  >
    <img
      alt="logo"
      class=""
      src="ui.png"
      style="width: 44px; height: 44px; filter: invert(1);"
    />
    <h1
      class="h1"
    >
      My Custom 
      <br />
      <span
        class="span"
      >
        UI Design Components
      </span>
    </h1>
  </a>
  <div>
    <h3>
      Props Documentation
    </h3>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th
            class="header"
          >
            Prop Name
          </th>
          <th
            class="header"
          >
            Type
          </th>
          <th
            class="header"
          >
            Description
          </th>
          <th
            class="header"
          >
            Modify
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="cell"
          >
            className
          </td>
          <td
            class="cell"
          >
            string
          </td>
          <td
            class="cell"
          >
            Classe CSS suppl√©mentaire pour personnaliser le style du logo. Optionnel.
          </td>
          <td
            class="cell"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
